### PR TITLE
Fixes (?) GitHub issue #30189 - Question about note on Creating and Throwing exceptions page

### LINKED ISSUE #30189: Question about note on Creating and Throwing exceptions page
The Creating and Throwing exceptions page https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/exceptions/creating-and-throwing-exceptions has a note that says:

"[!NOTE]The example above is for illustrative purposes. Index validating via exceptions is in most cases bad practice. Exceptions should be reserved to guard against exceptional program conditions, not for argument checking as above."

What is the reasoning for this note?  My understanding is that all .NET error handling should be done using exceptions.  Seems like this note is incomplete.

On this page it says to throw an exception instead of returning an error code.  What would be the best practice for returning argument checking information?

https://docs.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions#throw-exceptions-instead-of-returning-an-error-code

**Target framework**

- [x] .NET Core
- [x] .NET Framework
- [x] .NET Standard

<details>
<summary><strong>dotnet --info output</strong> or <strong>About VS info</strong></summary>

```console
<replace>
```
</details>

edit by @BillWagner add document block

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cfeeac2f-fbeb-b4b5-ff02-fd48a28a2768
* Version Independent ID: 4d98bec1-a03a-4a9c-4108-cb847f9c18bd
* Content: [Creating and Throwing Exceptions](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/exceptions/creating-and-throwing-exceptions)
* Content Source: [docs/csharp/fundamentals/exceptions/creating-and-throwing-exceptions.md](https://github.com/dotnet/docs/blob/main/docs/csharp/fundamentals/exceptions/creating-and-throwing-exceptions.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by v-thepet**
## Summary

Describe your changes here.

Fixes #30189

Edits article [Create and Throw Exceptions](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/exceptions/creating-and-throwing-exceptions)

Note: The issue says that the code sample should be updated, but the original issue is to update the note.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/fundamentals/exceptions/creating-and-throwing-exceptions.md](https://github.com/dotnet/docs/blob/2dbc395a4f2ebfba8ddcfe6a985f2bc4a301a0fa/docs/csharp/fundamentals/exceptions/creating-and-throwing-exceptions.md) | [Create and throw exceptions](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/exceptions/creating-and-throwing-exceptions?branch=pr-en-us-37822) |

### CODE REVIEW

File: docs/csharp/fundamentals/exceptions/creating-and-throwing-exceptions.md [hunk 1]
```diff
@@ -26,8 +26,8 @@ Programmers should throw exceptions when one or more of the following conditions
 
   :::code language="csharp" source="snippets/exceptions/Program.cs" ID="InvalidArg":::
 
-> [!NOTE]
-> The example above is for illustrative purposes. Index validating via exceptions is in most cases bad practice. Exceptions should be reserved to guard against exceptional program conditions, not for argument checking as above.
+  > [!NOTE]
+  > The preceding example is for illustrative purposes. Ideally, you don't use exceptions to validate arguments, but to handle exceptional program conditions.
```
Comment (BillWagner) [inline docs/csharp/fundamentals/exceptions/creating-and-throwing-exceptions.md:30]: I think we should reword this. Consider this change:

```suggestion
> [!NOTE]
> The preceding example shows how to use the `InnerException` property. It's intentionally simplified. In practice, you'd check that an index is in range before using it. You'd use this technique of wrapping an exception when a member of a parameter throws an exception you couldn't anticipate before calling the member.
```

**Review by BillWagner — COMMENTED** [review]
Hi @v-thepet 

Thanks for the questions. I offered a different suggestion. Feel free to accept it or offer a different wording.

File: docs/csharp/fundamentals/exceptions/creating-and-throwing-exceptions.md (hunk 1 above)
Comment (v-thepet) [reply docs/csharp/fundamentals/exceptions/creating-and-throwing-exceptions.md]: How about - to remove future tense:
  > The preceding example shows how to use the `InnerException` property. It's intentionally simplified. In practice, you should check that an index is in range before using it. You could use this technique of wrapping an exception when a member of a parameter throws an exception you couldn't anticipate before calling the member.

Comment (BillWagner) [reply docs/csharp/fundamentals/exceptions/creating-and-throwing-exceptions.md]: I like it. I updated my suggestion to match your edits.

**Review by BillWagner — APPROVED** [review]

**Pull request merged**